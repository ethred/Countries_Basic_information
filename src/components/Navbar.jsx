import './navbar.module.css';

function Navbar() {
  const date = new Date();
  return (
    <nav>
      <h3>
        Â©
        {date.getFullYear()}
      </h3>
      <h1>Countries Of The World</h1>
      <div className="icons">
        <img src="https://img.icons8.com/ios/30/microphone.png" alt="microphone icon" />
        <img src="https://img.icons8.com/30/settings--v1.png" alt="settings icon" />
      </div>
    </nav>
  );
}

export default Navbar;
