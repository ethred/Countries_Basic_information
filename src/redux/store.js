import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
