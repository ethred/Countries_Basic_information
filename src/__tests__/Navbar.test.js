import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../components/Navbar';
import store from '../redux/store';

test('Check navbar component rendered correctly', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
