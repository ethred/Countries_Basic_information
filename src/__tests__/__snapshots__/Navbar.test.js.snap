// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the CountryDetail component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3>
          ©
          2023
        </h3>
        <h1>
          Countries Of The World
        </h1>
        <div
          class="icons"
        >
          <img
            alt="microphone icon"
            src="https://img.icons8.com/ios/30/microphone.png"
          />
          <img
            alt="settings icon"
            src="https://img.icons8.com/30/settings--v1.png"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3>
        ©
        2023
      </h3>
      <h1>
        Countries Of The World
      </h1>
      <div
        class="icons"
      >
        <img
          alt="microphone icon"
          src="https://img.icons8.com/ios/30/microphone.png"
        />
        <img
          alt="settings icon"
          src="https://img.icons8.com/30/settings--v1.png"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check if the CountryDetail component rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3>
          ©
          2023
        </h3>
        <h1>
          Countries Of The World
        </h1>
        <div
          class="icons"
        >
          <img
            alt="microphone icon"
            src="https://img.icons8.com/ios/30/microphone.png"
          />
          <img
            alt="settings icon"
            src="https://img.icons8.com/30/settings--v1.png"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3>
        ©
        2023
      </h3>
      <h1>
        Countries Of The World
      </h1>
      <div
        class="icons"
      >
        <img
          alt="microphone icon"
          src="https://img.icons8.com/ios/30/microphone.png"
        />
        <img
          alt="settings icon"
          src="https://img.icons8.com/30/settings--v1.png"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check navbar component rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h3>
          ©
          2023
        </h3>
        <h1>
          Countries Of The World
        </h1>
        <div
          class="icons"
        >
          <img
            alt="microphone icon"
            src="https://img.icons8.com/ios/30/microphone.png"
          />
          <img
            alt="settings icon"
            src="https://img.icons8.com/30/settings--v1.png"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h3>
        ©
        2023
      </h3>
      <h1>
        Countries Of The World
      </h1>
      <div
        class="icons"
      >
        <img
          alt="microphone icon"
          src="https://img.icons8.com/ios/30/microphone.png"
        />
        <img
          alt="settings icon"
          src="https://img.icons8.com/30/settings--v1.png"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
